docs/src/main/asciidoc/config-reference.adoc

```diff
@@ -0,0 +1,772 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Configuration Reference Guide
+
+include::./attributes.adoc[]
+
+:numbered:
+:sectnums:
+:sectnumlevels: 4
+:toc:
+
+In this reference guide we're going to describe various aspects of Quarkus configuration.
+A Quarkus application and Quarkus itself (core and extensions) are both configured via the same mechanism that leverages the https://microprofile.io/project/eclipse/microprofile-config[MicroProfile Config] API.
+Quarkus configuration is based on https://github.com/smallrye/smallrye-config[SmallRye Config], an implementation of the MicroProfile Config specification.
+All of the standard features are supported.
+Moreover, there are several additional features which are made available by the SmallRye Config project as well as by Quarkus itself.
+
+TIP: If you're looking for information how to make a Quarkus extension configurable then see the <<writing-extensions.adoc#configuration,Writing Your Own Extension>> guide.
+
+[[configuration_sources]]
+== Configuration Sources
+
+By default, Quarkus reads configuration properties from several sources (in decreasing priority):
+
+1. System properties
+2. Environment variables
+3. A file named `.env` placed in the current working directory
```

Comment:
It should probably be noted that this file is used to determine properties using ENV VAR naming rules, not regular property naming rules.

Reply:
Yes, it is mentioned below in the detail section. This list should be short and concise. Maybe we should make all list items links to relevant detail sections?

Reply:
I like the idea